src/main/kotlin/org/jetbrains/teamcity/rest/implementation.kt

```diff
@@ -263,6 +269,30 @@ private class ParameterImpl(private val bean: ParameterBean) : Parameter {
         get() = bean.own
 }
 
+private class FinishBuildTriggerImpl(private val bean: TriggerBean) : FinishBuildTrigger {
+    override val initiatedBuildConfiguration: BuildConfigurationId
+        get() = BuildConfigurationId(bean.properties?.property?.find { it.name == "dependsOn" }?.value!!)
+
+    override val afterSuccessfulBuildOnly: Boolean
+        get() = bean.properties?.property?.find { it.name == "afterSuccessfulBuildOnly" }?.value?.toBoolean() ?: false
+
+    override val branchFilter: List<BranchRule>
+        get() = bean.properties
+                    ?.property
+                    ?.find { it.name == "branchFilter" }
+                    ?.value
+                    ?.split(" ")
+                    ?.map { BranchRuleImpl(it) }.orEmpty()
+}
+
+internal class BranchRuleImpl(private val branchRule: String) : BranchRule {
+    override val include: Boolean
+        get() = !branchRule.startsWith("-:")
+
+    override val branchPattern: String
+        get() = branchRule.substringAfter(":")
```

Comment:
':' isn't allowed in branch name, please check this [article](https://git-scm.com/docs/git-check-ref-format). "+:" suffix is also possible, so we have to support both cases.